src/herbpy/action/stacking.py

```diff
@@ -0,0 +1,66 @@
+import prpy, numpy
+from prpy.action import ActionMethod
+
+@ActionMethod
+def StackCups(robot, table, cup, stack, cups_stacked):
+    """
+    @param robot The robot performing the stacking
+    @param cup The cup that is being stacked
+    @param stack The stack of cups the cup is going to be placed on
+    @param cups_stacked A list of the cups that have been stacked 
+    """
+	#Grasp cup
+    env = robot.GetEnv()
+    robot.PushGrasp(cup, yaw_range=[0, numpy.pi])
+
+    with env:
+        aabb_cup = cup.ComputeAABB()
+        aabb_stack = stack.ComputeAABB()
+        stack_height = 2*aabb_stack.extents()[2] + (len(cups_stacked)+1)*0.03
+        manip = robot.GetActiveManipulator()
+
+    from prpy.rave import Disabled
+    with Disabled(table):
+        #Lift up cup
+        manip.PlanToEndEffectorOffset(direction=[0, 0, 1], distance = stack_height, position_tolerance=0.1, execute=True)
+
+    #Move cup in x-direction
+    move_cup_x, x_direc = MoveCupDistAndDirec(cup, stack, 0);
+    manip.PlanToEndEffectorOffset(direction=[x_direc, 0, 0], distance = move_cup_x, position_tolerance=0.1, execute=True)
```

Comment:
I don't think it's safe to assume the cups can be moved purely in the `x` and `y` axes. Shouldn't the direction be the vector from the cup to the stack in the world frame?